packages/subgraph/src/mappingHelpers.ts

```diff
@@ -523,19 +523,31 @@ export function updateAggregateIDASubscriptionsData(
  */
 function updateATSBalanceAndUpdatedAt(
     accountTokenSnapshot: AccountTokenSnapshot,
-    block: ethereum.Block
+    block: ethereum.Block,
+    balanceDelta: BigInt | null
 ): AccountTokenSnapshot {
     let superTokenContract = SuperToken.bind(
         Address.fromString(accountTokenSnapshot.token)
     );
-    let newBalanceResult = superTokenContract.try_realtimeBalanceOf(
-        Address.fromString(accountTokenSnapshot.account),
-        block.timestamp
-    );
-    if (!newBalanceResult.reverted) {
+
+    if (balanceDelta && accountTokenSnapshot.totalSubscriptionsWithUnits == 0) {
         accountTokenSnapshot.balanceUntilUpdatedAt =
-            newBalanceResult.value.value0;
+            accountTokenSnapshot.balanceUntilUpdatedAt.plus(balanceDelta as BigInt);
+    } else {
+        // if the account has any subscriptions with units we assume that
+        // the balance data requires a RPC call for balance because we did not
+        // have claim events there and we do not count distributions
+        // for subscribers
+        let newBalanceResult = superTokenContract.try_realtimeBalanceOf(
```

Comment:
piggybacked changes? we had issue with this?

Reply:
yeah, the balanceOf stuff is piggybacked changes - there were no issues, but this should speed up indexing.